jsonforms-editor/src/editor/components/preview/ReactMaterialPreview.tsx

```diff
@@ -11,20 +11,21 @@ import {
   materialRenderers,
 } from '@jsonforms/material-renderers';
 import { JsonForms } from '@jsonforms/react';
-import React from 'react';
+import React, { useMemo } from 'react';
 
+import { generateEmptyData } from '../../../core/model';
 import { useExportSchema, useExportUiSchema } from '../../../core/util/hooks';
 import { previewOptions } from './options';
 
 export const ReactMaterialPreview: React.FC = () => {
   const schema = useExportSchema();
   const uischema = useExportUiSchema();
-
+  const previewData = useMemo(() => generateEmptyData(schema), [schema]);
```

Comment:
`generateEmptyData` expects our custom `SchemaElement` type but it gets the exported JSON Schema. This works by chance because our custom `SchemaElement` also has a `type` which will be set to `object` for objects ;)

This is easily overlooked as the `schema` type is `any` here. We should `buildJsonSchema` in `schema.ts` to return ` JsonSchema`, then Typescript would complain about the mismatched types here.

I would like to suggest the following:
1. Adapt the type of `buildJsonSchema` and other methods in `schema.ts` which are not correctly typed.
2. Adapt the `useExportSchema` custom hook to take an optional boolean parameter. If it's set to true it should also return the original schema, e.g. `{schema, exportSchema}`, otherwise it should just return the exportSchema
3. Do the same for `useExportUiSchema` for consistency.
4. Use the new hook variants in the previews to calculate the previewData and to set the correct `schema`.
5. (Optional but recommended) remove the `useTransform` and `useEffectAfterInit` custom hooks and just replace it with `useMemo`. They are no longer needed.

Reply:
Ah, I missed that detail. I fixed it now and added types to the methods from `schems.ts`. However, I kept the `useExportSchema` hook as it is: in all other cases where the export schema was needed, the original one was not. And in case the original schema is needed, it's readily available via the `useSchema` hook. Modifying the `useExportSchema` and `useExportUiSchema` hooks to return the original schema on request only leads to type checking of the result everywhere the hooks are used. 
But if you feel strongly about it, I can change it.